Source/SwiftLintFramework/Models/ConfigurationError.swift

```diff
@@ -0,0 +1,13 @@
+//
+//  ConfigurationError.swift
+//  SwiftLint
+//
+//  Created by Scott Hoyt on 1/19/16.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+
+public enum ConfigurationError: ErrorType {
```

Comment:
For the time being, I made this *marginally* more useful by not squashing the error in `Configuration.rulesFromDict`, though nothing is really being done with it. Let me know if you want to combine this with `YamlParserError`. I like that we have one less error type with that approach. But I don't like that a `ConfigProviderRule` would even know about something that knows about the `Yaml` subsystem.